app/graphql/mutations/profile/associate_rules.rb

```diff
@@ -9,15 +9,16 @@ class AssociateRules < BaseMutation
       graphql_name 'associateRules'
 
       argument :id, ID, required: true
-      argument :rule_ids, [ID], required: true
+      argument :rule_ids, [ID], required: false
+      argument :rule_ref_ids, [ID], required: false
       field :profile, Types::Profile, null: true
 
       def resolve(args = {})
         ::Profile.transaction do
           profile = find_profile(args[:id])
           if profile
             rules_added, rules_removed = profile.update_rules(
-              ids: args[:rule_ids]
+              prepare_params(args)
```

Comment:
Observation: The param checking might be called earlier, before the transaction.